### PR TITLE
Add enter_maintenance and exit_maintenance roles

### PR DESCRIPTION
**Pull request by markgoddard**

### CODE REVIEW

File: roles/enter_maintenance/tasks/enter.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+---
+- name: Check if host can enter maintenance mode
+  ansible.builtin.import_role:
+    name: stackhpc.cephadm.commands
+  vars:
+    cephadm_commands:
+      - "orch host ok-to-stop {{ ansible_facts.nodename }}"
```
Comment (jackhodgkiss) [inline roles/enter_maintenance/tasks/enter.yml:7]: Do we need to be careful for RGW? The command `orch host ok-to-stop` doesn't check consider RGW. https://docs.ceph.com/en/reef/dev/cephadm/host-maintenance/#the-check-option

Comment (markgoddard) [reply roles/enter_maintenance/tasks/enter.yml:7]: Possibly. I'm not sure what we would do though. I don't currently have an environment in which to test it with RGW.

Comment (markgoddard) [reply roles/enter_maintenance/tasks/enter.yml:7]: I don't see the check option mentioned there.

Comment (jackhodgkiss) [reply roles/enter_maintenance/tasks/enter.yml:7]: I think the documentation is a bit misleading and that `--check` doesn't exist as a flag but rather as a way to refer to `orch host ok-to-stop`.

As for running `orch host ok-to-stop` on a host with RGW it produces the following.

```
WARNING: Removing RGW daemons can cause clients to lose connectivity.
ALERT: Cannot stop active Mgr daemon, Please switch active Mgrs with 'ceph mgr fail ceph-mgr-01'
```

Comment (cityofships) [reply roles/enter_maintenance/tasks/enter.yml:7]: `orch host maintenance enter` will never work without `--force` when there are RGWs - sadly.

File: roles/enter_maintenance/tasks/enter.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    - name: Extract full name of active Ceph manager
+      ansible.builtin.set_fact:
+        active_ceph_mgr: "{{ cephadm_commands_result.results[0].stderr | split | last | replace(\"'\", '') }}"
+
+    - name: Ensure active manager has been switched to another node
+      ansible.builtin.import_role:
+        name: stackhpc.cephadm.commands
+      vars:
+        cephadm_commands:
+          - "mgr fail {{ active_ceph_mgr }}"
```
Comment (mnasiadka) [inline roles/enter_maintenance/tasks/enter.yml:32]: {{ active_ceph_mgr }} is rather not needed - "ceph mgr fail" should work

Comment (markgoddard) [reply roles/enter_maintenance/tasks/enter.yml:32]: Ack

File: roles/enter_maintenance/tasks/enter.yml (hunk 1 above)
Comment (markgoddard) [reply roles/enter_maintenance/tasks/enter.yml:7]: Looks like we can remove the `rgw` label from the host temporarily:
```
ceph orch host label rm <host> rgw
# wait for rgw daemon to stop
ceph orch host ok-to-stop <host>
ceph orch host maintenance enter <host>
ceph orch host label add <host> rgw
```

Comment (markgoddard) [reply roles/enter_maintenance/tasks/enter.yml]: Done

**Review by mnasiadka — APPROVED** [review]

File: roles/enter_maintenance/tasks/enter.yml [hunk 3]
```diff
@@ -0,0 +1,68 @@
+---
+- name: Check if host can enter maintenance mode
+  ansible.builtin.include_role:
+    name: stackhpc.cephadm.commands
+  vars:
+    cephadm_commands:
+      - "orch host ok-to-stop {{ cephadm_hostname }}"
+
+# Annoyingly, 'ceph orch host ok-to-stop' does not exit non-zero when
```
[the remark this answers is not in the document]
Comment (markgoddard) [reply roles/enter_maintenance/tasks/enter.yml:9]: I'll take a look.

**Pull request merged**